### PR TITLE
feat: add auth args when helm repo add/update

### PR DESCRIPTION
**Pull request by zqq454224016**
feat: add auth args when helm repo add/update

Fix: #113

### CODE REVIEW

File: pkg/helm/repo.go [hunk 1]
```diff
@@ -81,8 +81,8 @@ func allProviders(settings *cli.EnvSettings, httpRequestTimeout time.Duration) g
 
 // RepoAdd
 // inspire by https://github.com/helm/helm/blob/dbc6d8e20fe1d58d50e6ed30f09a04a77e4c68db/cmd/helm/repo_add.go
-func RepoAdd(ctx context.Context, logger logr.Logger, name, url string, httpRequestTimeout time.Duration) (err error) {
-	entry := repo.Entry{Name: name, URL: url} // TODO add auth args
+func RepoAdd(ctx context.Context, logger logr.Logger, name, url, username, password string, httpRequestTimeout time.Duration) (err error) {
+	entry := repo.Entry{Name: name, URL: url, Username: username, Password: password} // TODO add auth args
```
Comment (0xff-dev) [inline pkg/helm/repo.go:85]: delete TODO comment.

File: pkg/repository/chartmuseum.go [hunk 2]
```diff
@@ -99,7 +102,17 @@ type chartmuseum struct {
...
 	_ = helm.RepoRemove(c.ctx, c.logger, c.repoName)
-	if err := helm.RepoAdd(c.ctx, c.logger, c.repoName, c.instance.Spec.URL, c.duration/2); err != nil {
+	if c.instance.Spec.AuthSecret != "" {
+		i := v1.Secret{}
+		if err := c.c.Get(c.ctx, types.NamespacedName{Name: c.instance.Spec.AuthSecret, Namespace: c.instance.GetNamespace()}, &i); err != nil {
+			c.logger.Error(err, "Failed to get secret")
+			return
+		}
+		c.username = string(i.Data["username"])
+		c.password = string(i.Data["password"])
```
Comment (0xff-dev) [inline pkg/repository/chartmuseum.go:112]: `username` and `password` have constant definitions here.
https://github.com/kubebb/core/blob/542ef80fde088f929800bcc15eb8ff7b695a4fb0/api/v1alpha1/repository.go#L26

File: pkg/repository/chartmuseum.go [hunk 3]
```diff
@@ -56,6 +56,7 @@ func NewChartmuseum(
 	instance *v1alpha1.Repository,
 	cancel context.CancelFunc,
 ) IWatcher {
+	//REPO  in
```
Comment (0xff-dev) [inline pkg/repository/chartmuseum.go:59]: delete comment.

**Comment by Abirdcfly** [comment]
**some git tips:https://submariner.io/development/code-review/#address-code-review-feedback-with-new-commits**

maybe add tests into config/sample/examples.sh? Any ideas? @0xff-dev @bjwswang

**Pull request closed without merging**